src/physics/initial_conditions/initial_condition_function.cpp

```diff
@@ -67,7 +104,17 @@ ::convert_primitive_to_conversative_value(
     const dealii::Point<dim,real> &point, const unsigned int istate) const
 {
     real value = 0.0;
-    if (dim == 3) {
+    if constexpr(dim == 3) {
+        std::array<real,nstate> soln_primitive;
+
+        soln_primitive[0] = primitive_value(point,0);
+        soln_primitive[1] = primitive_value(point,1);
+        soln_primitive[2] = primitive_value(point,2);
+        soln_primitive[3] = primitive_value(point,3);
+        soln_primitive[4] = primitive_value(point,4);
+
+        value = this->convert_primitive_to_conversative_euler(soln_primitive, this->gamma_gas, istate);
+/*
         const real rho = primitive_value(point,0);
         const real u   = primitive_value(point,1);
```

Comment:
Feel free to just delete this commented code.

Reply:
Done, sorry for committing that.